src/Contracts/PhpSoap/Client.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Contracts\PhpSoap;
+
+use SoapHeader;
+
+interface Client
+{
+    public function setEndpoint(string $endpoint): static;
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function setOptions(array $options): static;
+
+    /**
+     * Set the given headers on the request.
+     *
+     * @param array<int, SoapHeader> $headers
+     */
+    public function setHeaders(array $headers): static;
+
+    /**
+     * Make a request and return its response.
+     */
+    public function call(string $method, mixed $body): mixed;
```

Comment:
maybe better to change `mixed $body` to `array $arguments`?
That way it is compatible with how ext-soap works internally and with `php-soap/engine`:
 
- https://www.php.net/manual/en/soapclient.call.php
- https://github.com/php-soap/engine/blob/main/src/Engine.php

It allows for multi-params requests (see #45)